log-server/server.py

```diff
@@ -0,0 +1,30 @@
+from flask import Flask, request, jsonify
+
+app = Flask(__name__)
+
+# Define log file path
+log_file = "logs.txt"
+
+@app.route('/logs', methods=['POST'])
+def receive_logs():
+  print("Received a POST request to /logs")  # Added for debugging
+  # Check if request is JSON
+  if not request.is_json:
+    return jsonify({'error': 'Request must be JSON'}), 400
+
+  # Get the JSON data (assuming an array of log messages)
+  data = request.get_json()
+
+  # Open the log file in append mode
```

Comment:
for future, remember there can be many requests that are coming in parallel and you cannot open the same file for all of them since they will not be flushed to the disk correctly. Best is to open the file and leverage the file handler in each request processing so you don't have too many file descriptors and avoid locking related errors.